mirrord/kube/src/error.rs

```diff
@@ -0,0 +1,48 @@
+use thiserror::Error;
+
+pub type Result<T, E = KubeApiError> = std::result::Result<T, E>;
+
+#[derive(Debug, Error)]
+pub enum KubeApiError {
+    #[error("mirrord-layer: Kube failed with error `{0}`!")]
```

Comment:
I'm thinking of renaming it from `KubeApi` to something else like `AgentApi` or something of this sort because it is not only for the kubernetes but connection too